ConfigurationTool/Checks/Attributes.cs

```diff
@@ -47,7 +47,7 @@ public async Task GetAttributeChunkSizes(CancellationToken token)
 
             if (Session == null || !Session.Connected)
             {
-                await Run(token);
+                await Run(token, 0);
```

Comment:
What does 0 mean here? If 0 is timeout isn't it too short for session establishing purpose?

Reply:
Also shouldn't it be a configurable option to adjust to different needs?

Reply:
Hm. There is no real advantage to timing out reconnects for the extractor, since it restarts on crash anyway. I added this because we rely so heavily on the Run method failing in tests. I.e. letting it be 0 for tests, and leaving it at -1 for normal operation is fine.

The logic behind 0 is that the timeout is checked _after_ a failed reconnect attempt. So it tries to connect, and if it fails it checks if `timeout` has elapsed, and rethrows the exception if it has. So 0 just means always rethrow the exception.